TeachingRecordSystem/src/TeachingRecordSystem.Api/V3/Implementation/Operations/SetProfessionalStatus.cs

```diff
@@ -0,0 +1,606 @@
+using System.Diagnostics;
+using Medallion.Threading;
+using Microsoft.Xrm.Sdk.Query;
+using TeachingRecordSystem.Api.Infrastructure.Security;
+using TeachingRecordSystem.Api.V3.Implementation.Dtos;
+using TeachingRecordSystem.Core.DataStore.Postgres;
+using TeachingRecordSystem.Core.DataStore.Postgres.Models;
+using TeachingRecordSystem.Core.Dqt;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
+using TeachingRecordSystem.Core.Services.DqtOutbox;
+using TeachingRecordSystem.Core.Services.DqtOutbox.Messages;
+using TeachingRecordSystem.Core.Services.TrsDataSync;
+using ProfessionalStatusStatus = TeachingRecordSystem.Api.V3.Implementation.Dtos.ProfessionalStatusStatus;
+
+namespace TeachingRecordSystem.Api.V3.Implementation.Operations;
+
+public record SetProfessionalStatusCommand(
+    string Trn,
+    string SlugId,
```

Comment:
Don't want slug in these new operations - that's not a thing in TRS. Should be `SourceApplicationUserId` and `SourceApplicationReference`.